core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/StructuredEventFilter.java

```diff
@@ -236,41 +244,72 @@ private void putResourceIdFromRepository(ContainerRequestContext requestContext,
     }
 
     protected void extendRestParamsFromResponse(Map<String, String> params, CharSequence responseBody) {
-        if (responseBody != null && isResourceIdIsAbsentOrNull(params)) {
-            String resourceId = extractResourceIdFromJson(responseBody);
-            if (StringUtils.isEmpty(resourceId)) {
-                Matcher matcher = extendRestParamsFromResponsePattern.matcher(responseBody);
-                if (matcher.find() && matcher.groupCount() >= 1) {
-                    resourceId = matcher.group(1);
-                }
-            }
+        boolean resourceIdIsAbsentOrNull = isResourceIdIsAbsentOrNull(params);
+        boolean resourceCrnIsAbsentOrNull = isResourceCrnIsAbsentOrNull(params);
+        if (responseBody != null && (resourceIdIsAbsentOrNull || resourceCrnIsAbsentOrNull)) {
+            Set<String> resourcesParamsToCollect = new HashSet<>();
+            decorateSetWithValueIfNecessary(resourceIdIsAbsentOrNull, ID, resourcesParamsToCollect);
+            decorateSetWithValueIfNecessary(resourceCrnIsAbsentOrNull, CRN, resourcesParamsToCollect);
+            Map<String, String> resourceParams = extractResourceIdFromJson(responseBody, resourcesParamsToCollect);
```

Comment:
`extractResourceIdFromJson`: is it only extracting the resourceId, or other params as well?

Reply:
thanks, I forgot to rename